### PR TITLE
PLNSRVCE-1323: Set release mode and remove all proxies trust

### PR DESCRIPTION
**Pull request by enarha**
Set `release` (production) mode.
Set the list of trusted proxies to nil. By default gin is trusting all proxies and when running in debug mode the following warning is logged on start "You trusted all proxies, this is NOT safe. We recommend you to set a value". For more information:
https://pkg.go.dev/github.com/gin-gonic/gin#Engine.SetTrustedProxies

### CODE REVIEW

File: pkg/server/server.go
Comment (adambkaplan) [inline pkg/server/server.go:49]: We need to set the `TrustedPlatform` if we want to get the real client IP address. Otherwise by setting the trusted proxies to`nil`, the gin `context.ClientIP()` will likely return the IP of the AWS load balancer that sits in front of the OpenShift route.

```suggestion
	r.SetTrustedProxies(nil)
	r.TrustedPlatform = "X-Forwarded-For"
```

`X-Forwarded-For` is the standard forwarding header that AWS adds to their load balancers.

**Review by adambkaplan — CHANGES_REQUESTED** [review]

File: pkg/server/server.go
Comment (enarha) [reply pkg/server/server.go:49]: Updated.
For the record, the backend servers were still getting the original header including the "X-Forwarded-For" and were logging the IP from it. The difference is now the sprayproxy also logs the IP from the header instead of the original one. Thanks.

**Comment by adambkaplan** [comment]
/lgtm

@enarha how risky of a change do you think this is? Are we OK to merge it now since we are approaching a code freeze window?

**Comment by enarha** [comment]
I tested the change locally pretty thoroughly and it was working well (curl client and 2 gin servers as backends). I did not test with a request from GitHub and two backend clusters running PaC and that's the only risk I see. From the security point of view, if we merge the validation feature, that will get us covered, so no real urgency to get this one in. Bottom line is, I think we can wait.

**Comment by adambkaplan** [comment]
Let's wait until after Summit to merge this.

**Comment by adambkaplan** [comment]
@enarha are we tracking this change in JIRA somewhere?

**Comment by enarha** [comment]
No. I'll create one tomorrow.

**Comment by enarha** [comment]
PLNSRVCE-1323

**Comment by enarha** [comment]
/retest
retesting as we got the sast scanning enabled.

**Review by adambkaplan — APPROVED** [review]
lgtm

**Pull request merged**
